### DISCUSSION TITLE
editor.statusline position to show atleast 2 digits

### DISCUSSION THREAD

**Discussion by goyalyashpal** [helix-editor/helix#5776] [Ideas]
Hi!

- Is there some way to set the "column" in `position` in `[editor.statusline]` to always show _atleast_ two digits?
- as current behaviour makes the statusline extremely jittery while scrolling or navigating vertically

helix docs: [cursor position]
file: [position]

[cursor position]: https://docs.helix-editor.com/configuration.html?highlight=cursor%20position#editorstatusline-section
[position]: https://github.com/helix-editor/helix/blob/master/helix-core/src/position.rs

https://user-images.githubusercontent.com/19423063/216144044-ef4c5a53-4d74-44a2-811c-415d69337d03.mp4

----
### Additional details and thoughts

- If there's no way, then this can be either made optional (seems hard?) or be hard coded.
- with hard coding, it can be problematic for those dealing with files limited only to upto 9 columns
- ... like say todo lists, or csv data, or the like
- or those who use the jitter as an indication for something? but for them, using space would suffice 

<details>
<summary>Pedantic detailing, ignore if above description suffices</summary>
<br/>

In C language for print formatting:
- currently position shows: `"%d:%d", line, column`
- suggestion is to be: `"%d:%2d"` (use space) or `"%d:%02d"` (use zero)
- this would align the colon `:` hence solving jitter, for _most_ of the time

Currently it behaves as: (notice the colons don't align - thus jitter)
```
   1:1
  1:14
184:11
  24:1
```

Suggestion is to be:
```
  1: 1
  1:14
184:11
 24: 1
```

or
```
  1:01
  1:14
184:11
 24:01
```

</details>

**Comment by edrozenberg** [comment]
Same, would be great to be able to spec padding and 0-leader styling for various items. For ex. I set up my vim status line this way to have pretty much no status line stutter when moving through large files:

![Vim Statusline](https://github.com/user-attachments/assets/36ba6ef7-d6bd-4212-a3d8-f835f0a60e09)
